internal/provider/resource_aws_network_peering.go

```diff
@@ -66,14 +66,14 @@ func resourceAwsNetworkPeering() *schema.Resource {
 					return strings.ToLower(old) == strings.ToLower(new)
 				},
 			},
+			// Optional inputs
 			"peer_vpc_cidr_block": {
 				Description:  "The CIDR range of the peer VPC in AWS.",
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
```

Comment:
Can the API provide a default if the CIDR block is not set? We've run into issues in the past where combining `Optional` and `ForceNew` has led to unintended deletion of resources. That shouldn't be an issue here, since this was a required field and therefore should exist on all peering resources. Still, would be ideal to include `Computed` if the API can provide a default.

Reply:
This PR will result in the provider being in an interim state for these 2 resources, the eventual state that we are aiming for is releasing a major version that removes these 2 parameters altogether, delegating the creation of the routes to the HVN route APIs completely. At this moment the RPCs for create peering and create TGW attachment has stopped validating the existence of these parameters, and either 

(a) creates an HVN route for the peering/TGW attachment if `destination_cidr`/`peering_vpc_cidr_block` is present or 
(b) creates the TGW attachment and peering without an HVN route if `destination_cidr`/`peering_vpc_cidr_block` is not passed

I wasn't aware of how `ForceNew` could accidentally result in the deletion of an existing resource, in your opinion as the provider repo owner, do you think we should remove it for this interim state?

Reply:
As Ti mentioned, we are planning to deprecate/remove this field whatsoever (deprecate and eventually remove for peerings and remove right now for tgw attachments). We can use empty string as default if necessary.